src/features/common/Layout/UserPropsContext.tsx

```diff
@@ -98,36 +104,66 @@ export const UserPropsProvider: FC = ({ children }) => {
     logout({ returnTo: returnUrl });
   };
 
+  const fetchUserProfile = useCallback(
+    async (impersonationData?: ImpersonationData) => {
+      const header = {
+        'tenant-key': `${tenantConfig.id}`,
+        'X-SESSION-ID': await getsessionId(),
+        Authorization: `Bearer ${token}`,
+        'x-locale': locale,
+      };
```

Comment:
Explicitly define the return type for fetchUserProfile, and take care of any TS warnings that arise while doing so.